README.md

```diff
@@ -354,6 +354,37 @@ Checkout the new branch to continue working:
 (master)$ git checkout my-branch
 ```
 
+<a name="keep-whole-file"></a>
+### I want to keep the whole file from another ref-ish
+
+Say you have a working spike, with hundreds of changes. Everything is working. Now, you commit into another branch to save that work.
```

Comment:
@RichardLitt A term from XP:

> Development can also create spike solutions to analyze or solve the problem. These solutions are used for estimation and discarded once everyone gets clear visualization of the problem. Again, this may not influence the business requirements.

(from https://en.wikipedia.org/wiki/Extreme_programming_practices -> Release -> exploration phase)